backend/asmgen.ml

```diff
@@ -488,6 +488,11 @@ let build_asm_directives () : (module Asm_targets.Asm_directives_intf.S) = (
   )
 
 let emit_begin_assembly_with_dwarf ~disable_dwarf ~emit_begin_assembly ~sourcefile () =
+  if !Flambda_backend_flags.internal_assembler then
+    (X86_proc.register_internal_assembler Internal_assembler.assemble;
+    Emitaux.binary_backend_available := true;
+    Emitaux.create_asm_file := false)
```

Comment:
Where is `create_asm_file` used? I don't think it has any effect, see `Asmgen.compile_unit`. By the way, can the compiler create .s files with -internal-assembler?

Reply:
The `-S` option to create assembly files will still be supported, yes.